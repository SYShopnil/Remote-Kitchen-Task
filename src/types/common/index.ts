export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CIconWithHandlerButtonContainer = "CIconWithHandlerButtonContainer",
  CModal = "CModal",
  CProductAddSection = "CProductAddSection",
  CProductSearchBarContainer = "CProductSearchBarContainer",
  CForm = "CForm",
}
