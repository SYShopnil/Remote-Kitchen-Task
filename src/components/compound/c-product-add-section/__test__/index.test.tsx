import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { CProductAddSection } from "..";

describe("When Product Add section has rendered", () => {
  it("Expect it renders successfully", () => {
    const { getByTestId } = render(<CProductAddSection />);
    const myElement = getByTestId(EDataTestId.CProductAddSection);
    // const myElement = screen.getByTestId(EDataTestId.CIconWithHandlerButtonContainer);
    expect(myElement).toMatchSnapshot();
  });
});
