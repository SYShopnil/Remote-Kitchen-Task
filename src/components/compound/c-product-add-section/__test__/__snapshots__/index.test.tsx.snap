// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Product Add section has rendered Expect it renders successfully 1`] = `
<div
  data-testid="CProductAddSection"
>
  <div />
  <div>
    <span
      class="cursor-pointer"
    >
      <span
        class="flex justify-center items-center  "
      >
        <span
          class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EBF5EC] hover:bg-[#7F4D4F]  duration-[0.5s] text-[#7F4D4F] hover:text-[#EBF5EC]   border-[0.063rem] hover:border-[#EBF5EC] border-[#7F4D4F]   "
        >
          <span>
            Add++
          </span>
          <span
            class="ml-[1rem] h-[1.5rem] w-[1.5rem]  flex justify-center items-center"
          >
            <svg
              fill="#7F4D4F"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;
